src/common.ts

```diff
@@ -312,7 +312,17 @@ export class LoggingCommon {
     // Make sure that both callbacks are called in case if provided
     const newCallback: Callback = (err: Error | null, apiResponse?: {}) => {
       if (callback) {
-        callback(err, apiResponse);
+        try {
+          callback(err, apiResponse);
+        } catch (error) {
+          // We ignore here errors thrown from a callback calls (which is actually Writable.onwrite())
+          // since error could be temporary and following writes could succeed. We
+          // also want to make sure to call to defaultCallback suplied by a user, so
+          // user can get any details regarding error happened in defaultCallback
+          if (!this.defaultCallback) {
```

Comment:
One more potential solution

My main concern is around a specific edge case: where the callback throws a new exception, but `err` is still null, so nothing is shown to the user. From what you said, this should be impossible, but I'd feel better if we still added a guard against the impossible case, just to be safe.

What if we changed this function to something like this:

```
const newCallback: Callback = (err: Error | null, apiResponse?: {}) => {
      var callback_error = null
      if (callback) {
        try {
          callback(err, apiResponse);
        } catch (error) {
          callback_error = error
        }
      }
      if (this.defaultCallback) {
        this.defaultCallback(err, apiResponse);
      }
      if (callback_error && (!this.defaultCallback || err==null)){
          // we want to raise an error in two cases: if the defaultCallback isn't set, 
          // or if the callback throws a different error than the original call
          throw callback_error
      }
};
```


`if (callback_error && !this.defaultCallback)` should be equivalent to the existing logic. `if (callback_error && error==null)` shouldn't happen based on what you told me, but it makes sure no exceptions go unreported if our assumptions are broken

What do you think?

Reply:
Seems I came to same conclusion as you and worked on same idea for a fix :-) But I agree, this is a better way to handle the case.